python/pyarrow/pandas_compat.py

```diff
@@ -476,10 +478,25 @@ def convert_column(col, ty):
 
     types = [x.type for x in arrays]
 
+    if schema is not None:
+        # add index columns
+        index_types = types[len(column_names):]
+        for name, type_ in zip(index_column_names, index_types):
+            name = name if name is not None else 'None'
```

Comment:
It's not very convenient to have multiple columns named the same (e.g. "None"). It would be better to have a numbered fallback (e.g. "Column 3").

Reply:
I agree, but that's how it is currently done in other places as well (I needed to add the above to get the existing tests passing). 

(also, having multiple `None`-named columns in pandas should be a rare usecase)